# HG changeset patch
# User Carl Osterwisch <costerwisch@generalcompression.com>
# Date 1373565310 14400
#      Thu Jul 11 13:55:10 2013 -0400
# Node ID 22ec459cf7ba0bcaf9a6e975225d48a1a7943e6c
# Parent  ec6ced0ece879813c76c183919fc583ade799b1e
Correction to allow multi-line plots to have multiple colors (bug #16955).

diff --git a/scripts/plot/private/__next_line_color__.m b/scripts/plot/private/__next_line_color__.m
--- a/scripts/plot/private/__next_line_color__.m
+++ b/scripts/plot/private/__next_line_color__.m
@@ -39,16 +39,17 @@ function rgb = __next_line_color__ (rese
     ## Indicates whether the next call will increment or not
     reset_colors = reset;
   else
     ## Find and return the next line color
     ca = gca ();
     colorOrder = get (ca, "ColorOrder");
     if (reset_colors)
       color_index = 1;
+      reset_colors = false;
     else
       ## Executed when "hold all" is active
       nChildren = length (get (ca, "Children"));
       nColors = rows (colorOrder);
       color_index = mod (nChildren, nColors) + 1;
     endif
     rgb = colorOrder(color_index,:);
   endif
diff --git a/scripts/plot/private/__next_line_style__.m b/scripts/plot/private/__next_line_style__.m
--- a/scripts/plot/private/__next_line_style__.m
+++ b/scripts/plot/private/__next_line_style__.m
@@ -40,16 +40,17 @@ function [linestyle, marker] = __next_li
     ca = gca ();
     style_rotation = get (ca, "linestyleorder");
     if (ischar (style_rotation))
       style_rotation = strsplit (style_rotation, "|");
     endif
     nStyles = length (style_rotation);
     if (reset_style || (nStyles < 2))
       style_index = 1;
+      reset_style = false;
     else
       ## Executed when "hold all" is active
       nChildren = length (get (ca, "Children"));
       nColors = length (get (ca, "ColorOrder"));
       style_index = mod (floor (nChildren/nColors), nStyles) + 1;
     endif
     options = __pltopt__ ("__next_line_style__",
                           style_rotation(style_index));
